tests/lib/tools/tests.invariant

```diff
@@ -122,6 +123,22 @@ check_leftover_defer_sh() {
 	fi
 }
 
+check_broken_snaps() {
+	n="$1" # invariant name
+	(
+		# awk is hard - this just gives us the names of snaps that are broken
+		for brokesn in $(snap list --all | tail -n +2 | awk '{print $(NF) " " $1}' | grep 'broken ' | awk '{print $(NF)}' | sort | uniq); do
+			# could be multiple broken revisions
+			echo "snap $brokesn is broken"
+		done
+	) > "$TESTSTMP/tests.invariant.$n"
```

Comment:
Why write it to a file? Is this something we want to persist?

Reply:
no, but that's just how all the other invariants are implemented and I followed the pattern

Reply:
I see, thanks for clarifying